hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -232,7 +232,32 @@ extends Importer:
       term(rhs)
     case tree @ App(lhs, rhs) =>
       val sym = FlowSymbol("‹app-res›", nextUid)
-      Term.App(term(lhs), term(rhs))(tree, sym)
+      val lt = term(lhs)
+      val rt = term(rhs)
+
+      // Check if module arguments match module parameters
+      val args = rt match
+        case Term.Tup(fields) => S(fields)
+        case _ => N
+      val params = lt.symbol
+        .filter(_.isInstanceOf[BlockMemberSymbol])
+        .flatMap(_.asInstanceOf[BlockMemberSymbol].trmTree)
+        .filter(_.isInstanceOf[TermDef])
+        .flatMap(_.asInstanceOf[TermDef].paramLists.headOption)
```

Comment:
Methods `collect` + `flatten` can do this but safely. `asInstanceOf` should be avoided if possible.